src/dataflow-types/src/client/controller/storage.rs

```diff
@@ -372,19 +383,67 @@ where
 
             self.state.collections.insert(id, collection_state);
 
-            dataflow_commands.push(CreateSourceCommand {
-                id,
-                desc,
-                since,
-                storage_metadata: self.collection_metadata(id)?,
+            let storage_service = self
+                .orchestrator
+                .ensure_service(
+                    &id.to_string(),
+                    ServiceConfig {
+                        image: self.storaged_image.clone(),
+                        args: &|assigned| {
+                            vec![
+                                format!("--workers=1"),
+                                format!(
+                                    "--listen-addr={}:{}",
+                                    assigned.listen_host, assigned.ports["controller"]
+                                ),
+                                format!(
+                                    "--http-console-addr={}:{}",
+                                    assigned.listen_host, assigned.ports["http"]
+                                ),
+                            ]
+                        },
+                        ports: vec![
+                            ServicePort {
+                                name: "controller".into(),
+                                port_hint: 2100,
```

Comment:
Should this be `assigned.ports["controller"]`, and similarly for below?

Reply:
Ah, nope! `assigned` is only in scope on L420-429. The idea is that a service requests a particular port and you may or may not actually get that port. With the process orchestrator, for example, not every `storaged` process can get port 2100. But in Kubernetes since each pod is isolated you can hand out 2100 to every pod.